__test__/Ship.test.js

```diff
@@ -0,0 +1,63 @@
+const Ship = require('../src/Ship.js');
+
+describe('Ship', () => {
+    describe('with ports and an itinerary', () => {
+        let ship;
+        let dover;
+        let calais;
+        let itinerary;
+
+        beforeEach(() => {
+        dover = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Dover',
+            ships: []
+        };
+        calais = {
+            addShip: jest.fn(),
+            removeShip: jest.fn(),
+            name: 'Calais',
+            ships: []
+        };
+        itinerary = {
+            ports: [dover, calais]
+        };
+        ship = new Ship(itinerary);
+    });
+
+        it('can be instantiated', () => {
+        expect(ship).toBeInstanceOf(Object);
+    });
+
+    it('has a starting port', () => {
+        expect(ship.currentPort).toBe(dover);
+    });
+
+    it('can set sail', () => {
+        ship.setSail();
+      
+        expect(ship.currentPort).toBeFalsy();
+        expect(dover.removeShip).toHaveBeenCalledWith(ship);
+    });
+
+    it('can dock at a different port', () => {     
+        ship.setSail();
+        ship.dock();
+      
+        expect(ship.currentPort).toBe(calais);
+        expect(calais.addShip).toHaveBeenCalledWith(ship);
+    });
+
+    it('can\'t set sail further than its itinerary', () => {
+        ship.setSail();
+        ship.dock();
+
+        expect(() => ship.setSail()).toThrowError('End of itinerary reached');
+    });
+
+    it('gets added to port on instantiation', () => {
+        expect(dover.addShip).toHaveBeenCalledWith(ship);
+    });
```

Comment:
All of this needs pushing forward 4 spaces to bring it into line with the code above

The logic is fine, just needs moving about a bit!